apps/wallet/src/ui/app/pages/home/nft-details/index.tsx

```diff
@@ -1,179 +1,176 @@
-                        <NFTDisplayCard
-                            nftobj={nft}
-                            size="large"
-                            expandable={true}
-                        />
-                        {NFTDetails}
-                    </section>
-                </Content>
-                <Menu stuckClass={st.shadow} className={st.shadow}>
-                    <button
-                        onClick={onClick}
-                        className={cl(
-                            'btn',
-                            st.action,
-                            st.sendNftBtn,
-                            'primary'
-                        )}
-                        disabled={!isTransferable}
-                        title={
-                            isTransferable
-                                ? undefined
-                                : "Unable to send. NFT doesn't have public transfer method"
-                        }
-                    >
-                        <Icon
-                            icon={SuiIcons.ArrowLeft}
-                            className={cl(st.arrowActionIcon, st.angledArrow)}
-                        />
-                        Send NFT
-                    </button>
-                </Menu>
-            </BottomMenuLayout>
+            ))}
         </div>
     );
 }
 
 function NFTDetailsPage() {
     const [searchParams] = useSearchParams();
-    const [startNFTTransfer, setStartNFTTransfer] = useState<boolean>(false);
-    const [selectedNFT, setSelectedNFT] = useState<SuiObject | null>(null);
-    const objectId = useMemo(
-        () => searchParams.get('objectId'),
-        [searchParams]
+    const navigate = useNavigate();
+    const nftId = searchParams.get('objectId');
+    const nftSelector = useMemo(
+        () => createAccountNftByIdSelector(nftId || ''),
+        [nftId]
     );
-
-    const nftCollections = useAppSelector(accountNftsSelector);
-
-    const activeNFT = useMemo(() => {
-        const selectedNFT = nftCollections.filter(
-            (nftItem) => getObjectId(nftItem.reference) === objectId
-        )[0];
-        setSelectedNFT(selectedNFT);
-        return selectedNFT;
-    }, [nftCollections, objectId]);
-
-    const loadingBalance = useAppSelector(
-        ({ suiObjects }) => suiObjects.loading && !suiObjects.lastSync
-    );
-
-    const startNFTTransferHandler = useCallback(() => {
-        setStartNFTTransfer(true);
-    }, []);
-
-    if (!objectId || (!loadingBalance && !selectedNFT && !startNFTTransfer)) {
-        return <Navigate to="/nfts" replace={true} />;
-    }
-
+    const selectedNft = useAppSelector(nftSelector);
+    const isTransferable = !!selectedNft && hasPublicTransfer(selectedNft);
+    const shortAddress = useMiddleEllipsis(nftId);
+    const { nftFields, fileExtensionType, filePath } =
+        useNFTBasicData(selectedNft);
+    const { loading } = useObjectsState();
+    const detailAttrs = [
```

Comment:
I don't love APIs with config objects like this, I feel like this should just be props defined inline, or mapped or something

Reply:
moved them inline